CareKitStore/CareKitStore/CoreData/OCKStore+Outcomes.swift

```diff
@@ -151,6 +151,7 @@ extension OCKStore {
         currentOutcomes.forEach {
             $0.deletedDate = Date()
             $0.values = Set()
+            $0.logicalClock = Int64(context.clockTime)
```

Comment:
This one does seem necessary. I'd like to come up with a way to get around updating existing objects in a way that requires them to be synchronized a 2nd time, but I think it makes sense to include this, at least for the time being.

Reply:
The only way I can think of now is not creating a new tombstoned entity with a different uuid. This goes against the immutability of a record you mentioned before, but I always interpreted this as “most of parts of the record is immutable”. Since fields like the “logicalClock” is never mutated outside of the CareStore (nor is it exposed), it seems okay to me. I guess someone could argue then it’s not really immutable...